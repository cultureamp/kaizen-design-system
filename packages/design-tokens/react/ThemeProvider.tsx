import React from "react"
import { ThemeManager, defaultTheme, Theme } from "../"

export const ThemeContext = React.createContext<Theme>(defaultTheme)

/**
 * Wrap your application in this provider using a ThemeManager, to synchronise it with a react context.
 * This allows child react elements to more easily use theme variables, using the {@link useTheme} hook.
 */
export const ThemeProvider = ({
  themeManager,
  ...props
}: {
  themeManager: ThemeManager
  children: React.ReactNode
}) => {
  const [theme, setTheme] = React.useState<Theme>(
    themeManager.getCurrentTheme()
  )
  React.useEffect(() => {
    let cancelled = false
    const listener = (newTheme: Theme) => {
      if (!cancelled) setTheme(newTheme)
    }
    themeManager.addThemeChangeListener(listener)
    return () => {
      cancelled = true
      themeManager.removeThemeChangeListener(listener)
    }
  }, [])

  return (
    <ThemeContext.Provider value={theme}>
      {props.children}
    </ThemeContext.Provider>
  )
}

export const useTheme = () => React.useContext(ThemeContext)
