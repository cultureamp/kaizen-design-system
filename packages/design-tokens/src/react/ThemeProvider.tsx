import React, { useState } from "react"
import { ThemeManager, defaultTheme, Theme } from ".."

export const ThemeContext = React.createContext<Theme>(defaultTheme)

/**
 * Wrap your application in this provider using a ThemeManager, to synchronise it with a react context.
 * This allows child react elements to more easily use theme variables, using the {@link useTheme} hook.
 */
/**
 * @deprecated ThemeProvider is deprecated. Please use KaizenProvider from "@kaizen/components" instead.
 */
export const ThemeProvider = ({
  themeManager,
  ...props
}: {
  themeManager?: ThemeManager
  children: React.ReactNode
}): JSX.Element => {
  const [themeManagerInstance] = useState(
    () => themeManager || new ThemeManager(defaultTheme)
  )

  const [theme, setTheme] = React.useState<Theme>(
    themeManagerInstance.getCurrentTheme()
  )
  React.useEffect(() => {
    let cancelled = false
    const listener = (newTheme: Theme): void => {
      if (!cancelled) setTheme(newTheme)
    }
    themeManagerInstance.addThemeChangeListener(listener)
    return () => {
      cancelled = true
      themeManagerInstance.removeThemeChangeListener(listener)
    }
  }, [])

  return (
    <>
      <ThemeContext.Provider value={theme}>
        {props.children}
      </ThemeContext.Provider>
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-demi-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-medium.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-regular.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/tiempos/tiempos-headline-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
    </>
  )
}

export const useTheme = (): Theme => React.useContext(ThemeContext)
