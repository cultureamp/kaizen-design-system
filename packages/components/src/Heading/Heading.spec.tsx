import * as React from "react"
import { render } from "@testing-library/react"
import { AllowedHeadingTags, Heading, HeadingVariants } from "."

describe("<Heading />", () => {
  it("changes rendered HTML element when passed tag", () => {
    const headingMock = render(
      <Heading variant="display-0" tag="div">
        Example
      </Heading>
    )
    expect(headingMock.getByText("Example").tagName).toBe("DIV")
  })

  it("passes through data attributes", () => {
    const { getByTestId } = render(
      <Heading variant="display-0" data-testid="test-id">
        Example
      </Heading>
    )
    expect(getByTestId("test-id")).toBeInTheDocument()
  })

  describe("defaults to the correct HTML element", () => {
    type TestObject = { variant: HeadingVariants; el: AllowedHeadingTags }
    const testCases: TestObject[] = [
      { variant: "display-0", el: "h1" },
      { variant: "heading-1", el: "h1" },
      { variant: "heading-2", el: "h2" },
      { variant: "heading-3", el: "h3" },
      { variant: "heading-4", el: "h4" },
      { variant: "heading-5", el: "h5" },
      { variant: "heading-6", el: "h6" },
    ]

    testCases.forEach(({ variant, el }) => {
      it(`renders the correct element for <Heading variant={${variant}} />`, () => {
        const headingMock = render(<Heading variant={variant}>Example</Heading>)
        expect(headingMock.getByText("Example").tagName.toLowerCase()).toBe(el)
      })
    })
  })

  it("allows consumers to provide a className", () => {
    const { getByText } = render(
      <Heading variant="heading-4" classNameOverride="example-classname">
        Example
      </Heading>
    )
    expect(getByText("Example").classList).toContain("example-classname")
  })
})
