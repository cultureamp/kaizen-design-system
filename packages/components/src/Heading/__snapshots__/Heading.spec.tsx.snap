// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Heading /> > defaults to the correct HTML element > renders the correct element for <Heading variant={display-0} /> 1`] = `
<body>
  <div>
    <h1
      class="_heading_080c38 _display-0_080c38 _dark_080c38 _large_080c38"
    >
      Example
    </h1>
  </div>
</body>
`;

exports[`<Heading /> > defaults to the correct HTML element > renders the correct element for <Heading variant={heading-1} /> 1`] = `
<body>
  <div>
    <h1
      class="_heading_080c38 _heading-1_080c38 _dark_080c38 _large_080c38"
    >
      Example
    </h1>
  </div>
</body>
`;

exports[`<Heading /> > defaults to the correct HTML element > renders the correct element for <Heading variant={heading-2} /> 1`] = `
<body>
  <div>
    <h2
      class="_heading_080c38 _heading-2_080c38 _dark_080c38 _large_080c38"
    >
      Example
    </h2>
  </div>
</body>
`;

exports[`<Heading /> > defaults to the correct HTML element > renders the correct element for <Heading variant={heading-3} /> 1`] = `
<body>
  <div>
    <h3
      class="_heading_080c38 _heading-3_080c38 _dark_080c38 _small_080c38"
    >
      Example
    </h3>
  </div>
</body>
`;

exports[`<Heading /> > defaults to the correct HTML element > renders the correct element for <Heading variant={heading-4} /> 1`] = `
<body>
  <div>
    <h4
      class="_heading_080c38 _heading-4_080c38 _dark_080c38 _small_080c38"
    >
      Example
    </h4>
  </div>
</body>
`;

exports[`<Heading /> > defaults to the correct HTML element > renders the correct element for <Heading variant={heading-5} /> 1`] = `
<body>
  <div>
    <h5
      class="_heading_080c38 _heading-5_080c38 _dark_080c38 _small_080c38"
    >
      Example
    </h5>
  </div>
</body>
`;

exports[`<Heading /> > defaults to the correct HTML element > renders the correct element for <Heading variant={heading-6} /> 1`] = `
<body>
  <div>
    <h6
      class="_heading_080c38 _heading-6_080c38 _dark_080c38 _small_080c38"
    >
      Example
    </h6>
  </div>
</body>
`;
